app/src/main/java/org/fossasia/openevent/app/common/data/models/Event.java

```diff
@@ -112,4 +161,25 @@ public Event() { }
     private interface Excluding {
         List<Ticket> getEventTickets();
     }
+
+    public void setDefaults() {
+        paymentCountry = DEFAULT_STRING;
+        paypalEmail = DEFAULT_STRING;
+        paymentCurrency = DEFAULT_CURRENCY;
+        organizerDescription = DEFAULT_STRING;
+        onsiteDetails = DEFAULT_STRING;
+        organizerName = DEFAULT_STRING;
+        locationName = DEFAULT_STRING;
+        privacy = DEFAULT_STRING;
+        codeOfConduct = DEFAULT_STRING;
+        state = STATE_DRAFT;
+        searchableLocationName = DEFAULT_STRING;
+        description = DEFAULT_STRING;
+        bankDetails = DEFAULT_STRING;
+        chequeDetails = DEFAULT_STRING;
+        originalImageUrl = DEFAULT_PROTOCOL + File.separator + File.separator + DEFAULT_URL;
+        externalEventUrl = DEFAULT_PROTOCOL + File.separator + File.separator + DEFAULT_URL;
+        logoUrl = DEFAULT_PROTOCOL + File.separator + File.separator + DEFAULT_URL;
+        ticketUrl = DEFAULT_PROTOCOL + File.separator + File.separator + DEFAULT_URL;
+    }
```

Comment:
By single responsibility principle, this should not be in this class at all, this class is model which should be mostly untouched

Reply:
Also, don't initialize by default. These fields should be null if not provided by user

Reply:
When the fields are not initialized by default they are having null values which in turn causes NullPointerException in Dashboard which in turn shows blank. Also just after creating event I get a whte screen and app hangs though the event is created successfully. But on choosing that event in the event list layout, again the app hangs.  This is the error log for NullPointerException - 
`E/Logger: An exception occurred : value is null
          java.lang.NullPointerException: value is null
              at io.reactivex.internal.functions.ObjectHelper.requireNonNull(ObjectHelper.java:39)
              at io.reactivex.Single.just(Single.java:655)
              at org.fossasia.openevent.app.common.utils.core.CurrencyUtils.getCurrencySymbol(CurrencyUtils.java:44)
              at org.fossasia.openevent.app.module.main.MainPresenter.lambda$start$1$MainPresenter(MainPresenter.java:68)
              at org.fossasia.openevent.app.module.main.MainPresenter$$Lambda$2.accept(Unknown Source)
              at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
              at io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onNext(ObservableDoOnEach.java:103)
              at io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onNext(ObservableDoOnEach.java:103)
              at io.reactivex.internal.observers.DisposableLambdaObserver.onNext(DisposableLambdaObserver.java:58)
              at io.reactivex.subjects.PublishSubject$PublishDisposable.onNext(PublishSubject.java:264)
              at io.reactivex.subjects.PublishSubject.onNext(PublishSubject.java:183)
              at org.fossasia.openevent.app.common.data.Bus.pushSelectedEvent(Bus.java:19)
              at org.fossasia.openevent.app.module.event.list.EventsListAdapter$EventRecyclerViewHolder.lambda$new$0$EventsListAdapter$EventRecyclerViewHolder(EventsListAdapter.java:76)
              at org.fossasia.openevent.app.module.event.list.EventsListAdapter$EventRecyclerViewHolder$$Lambda$0.onClick(Unknown Source)
              at android.view.View.performClick(View.java:5612)
              at android.view.View$PerformClick.run(View.java:22285)
              at android.os.Handler.handleCallback(Handler.java:751)
              at android.os.Handler.dispatchMessage(Handler.java:95)
              at android.os.Looper.loop(Looper.java:154)
              at android.app.ActivityThread.main(ActivityThread.java:6123)
              at java.lang.reflect.Method.invoke(Native Method)
              at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
              at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)`

Reply:
Then ask for the payment currency from the user. And please create an issue to handle null currency type